modules/lang-painless/src/main/java/org/elasticsearch/painless/WriterConstants.java

```diff
@@ -40,22 +44,25 @@
 
     final static Type DEFINITION_TYPE = Type.getType(Definition.class);
 
+    final static Type OBJECT_TYPE = Type.getType(Object.class);
+    final static Method OBJECT_GETCLASS = getAsmMethod(Class.class, "getClass");
+
     final static Type MAP_TYPE  = Type.getType(Map.class);
     final static Method MAP_GET = getAsmMethod(Object.class, "get", Object.class);
 
     final static Type SCORE_ACCESSOR_TYPE    = Type.getType(ScoreAccessor.class);
     final static Method SCORE_ACCESSOR_FLOAT = getAsmMethod(float.class, "floatValue");
 
-    final static Method DEF_METHOD_CALL = getAsmMethod(
-        Object.class, "methodCall", Object.class, String.class, Definition.class, Object[].class, boolean[].class);
+    /** dynamic callsite bootstrap signature */
+    final static MethodType DEF_BOOTSTRAP_TYPE = MethodType.methodType(CallSite.class, MethodHandles.Lookup.class, 
+                                                                       String.class, MethodType.class, int.class);
+    final static Handle DEF_BOOTSTRAP_HANDLE = new Handle(Opcodes.H_INVOKESTATIC, "org/elasticsearch/painless/DynamicCallSite", 
```

Comment:
I'd use `Type.getInternalName(DynamicCallSite.class)` instead of constant. This allows to rename class if needed and not break static string constants